src/Symfony/Component/Routing/Loader/YamlFileLoader.php

```diff
@@ -127,6 +127,24 @@ protected function parseRoute(RouteCollection $collection, $name, array $config,
         $methods = isset($config['methods']) ? $config['methods'] : array();
         $condition = isset($config['condition']) ? $config['condition'] : null;
 
+        if (isset($requirements['_method'])) {
+            if (0 === count($methods)) {
+                $methods = explode('|', $requirements['_method']);
```

Comment:
@Tobion First, [requirements are set](https://github.com/symfony/symfony/blob/2.7/src/Symfony/Component/Routing/Route.php#L91). Then, [methods will be set](https://github.com/symfony/symfony/blob/2.7/src/Symfony/Component/Routing/Route.php#L99-101) if there are any, thus overriding previously set methods coming from the requirements.

I only set method here if none have been configured via the `methods` option (otherwise `count($methods)` would not be 0).